lib/types.ts

```diff
@@ -131,7 +139,44 @@ export interface EntryFieldAPI {
   getForLocale: (locale: string) => FieldAPI
 }
 
-export interface EntryAPI {
+type TaskState = 'active' | 'resolved'
+
+export interface TaskSys {
+  id: string
+  type: 'Task'
+  parentEntity: { sys: Link }
+  space: Link
+  environment: Link
+  createdBy: Link
+  createdAt: string
+  updatedBy: Link
+  updatedAt: string
+  version: number
+}
+
+export interface TaskInputData {
+  assignedToId: string
+  body: string
+  status: TaskState
+}
+
+export interface Task {
+  assignedTo: Link
+  body: string
+  status: TaskState
+  sys: TaskSys
+}
+
+/** Allows accessing the Task API for the current entry. */
+export interface TaskAPI {
+  getTask(id: string): Promise<Task>
+  getTasks(): Promise<CollectionResponse<Task>>
+  createTask(data: TaskInputData): Promise<Task>
+  updateTask(task: Task): Promise<Task>
+  deleteTask(task: Task): Promise<void>
```

Comment:
The delete calls in `SpaceAPI` seem to return the actual object that was deleted.
Any reason for not doing that here?

Reply:
I was going off of the CMA docs. This seems to be a mistake either in the docs, or in the SpaceAPI then, good catch! I'll investigate

Reply:
So this code is ok for now, but we'll need to change SpaceAPI later